packages/logger/README.md

```diff
@@ -0,0 +1,99 @@
+# Usage
+
+```js
+var Logger = require('@mongosh/logger')
+var path = require('path')
+var fs = require('fs')
+var os = require('os')
+
+var saveFilePath = path.join(os.homedir(), '.mongosh_log')
+var log = new Logger(saveFilePath)
+
+var infoObject = {
+  shellCommand: 'db.sales.find({ storeLocation: "Berlin" })'
+}
+
+var errorObject = {
+  shellCommand: 'db.sales.find({ storeLocation: "Berlin" })'
+  error: new TypeError()
+}
+
+log.info(infoObject)
+log.error(errorObject)
+```
+
+Logger uses [pino](https://github.com/pinojs/pino) under the hood for logging.
+
+Within mongosh architecture, `Logger` gets initialised along with the `Mapper`
+and requires passing in logger parameters to ShellMapper:
+
+![Using Logger with browser, Compass and CLI interfaces requires passing in
+logger parameters to ShellMapper](./logger-usage.png)
+
+```js
+var CliServiceProvider = require('mongosh-service-provider-server').CliServiceProvider
+var ShellApi = require('mongosh-shell-api')
+var Mapper = require('mongosh-mapper')
+
+var loggerPath = path.join(os.homedir(), '.mongosh_log')
+
+CliServiceProvider.connect(driverUri, driverOptions)
+  .then(function (serviceProvider) {
+    var mapper = new Mapper(serviceProvider, loggerPath)
+    var shellApi = new ShellApi(this.mapper)
+  })
+```
+
+## API
+### `new Logger([destination])`
```

Comment:
If we are going to enforce the constructor interface i'd prefer `new Logger(fileTransport)`, `new Logger(consoleTransport)`. 

We may also think to avoid that and just establish a contract between mapper and logger, then we can just instantiate `pino` or anything else that is appropriate, without over-constraining.

Reply:
i don't want to enforce constructor interface. These should work as well:

```js
var logger = require('@mongosh/logger')()
```
or:
```js
var logger = require('@mongosh/logger')(/path/to/dest/)
```
or:
```js
var logger = require('@mongosh/logger)({browser: {toObject: true}})
```
or:
```js
var logger = require('@mongosh/logger')
var log = logger(/path/to/file)
```
I'd prefer to not have to deal with any sort of transport layers, but instead provide defaults. `pino` handles both browser and node environments so we can piggy back on that.

What do you mean by "establish a contract"?

Reply:
Here I was still thinking we were trying to do only a Logger. In which case, as you mentioned somewhere else, I had the evil idea of just use `pino` and add  an interface (as in typescript Interface) in the mapper.